xc/device.ml

```diff
@@ -1870,15 +1870,20 @@ module Dm_Common = struct
     let get_pci_string = function
       | None -> raise (Xenopsd_error (Internal_error (Printf.sprintf "No PCI address (%s)" __LOC__)))
       | Some pci -> Xcp_pci.string_of_address pci in
+    let concat_two_strings str1 str2 =
+      match str1,str2 with
+      | str1,"" -> str1
+      | _,_ -> str1 ^ "," ^ str2 in
     let device_args =
       List.map (fun x ->
           match x.implementation with
           | Nvidia _conf ->
-             Printf.sprintf "--device=%s,%s,%s,%s"
+             _conf.extra_args |> concat_two_strings
+             (Printf.sprintf "--device=%s,%s,%s,%s"
```

Comment:
Please keep this simpler - it should be obvious from the format string, what arguments are passed with the `--device` option.

Reply:
Something like:
```
Printf.sprintf "--device=%s" (concat [
  (Xenops_interface.Pci.string_of_address x.physical_pci_address)
  _conf.type_id
  (get_pci_string _conf.virtual_pci_address)
  _conf.uuid
  _conf.extra_args
])
```